x-pack/plugins/apm/common/search_strategies/failed_transactions_correlations/types.ts

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { RawResponseBase, SearchStrategyClientParams } from '../types';
+
+import { FAILED_TRANSACTIONS_IMPACT_THRESHOLD } from './constants';
+
+export interface FailedTransactionsCorrelation {
+  key: string;
+  doc_count: number;
+  bg_count: number;
+  score: number;
+  pValue: number | null;
+  fieldName: string;
+  fieldValue: string;
+  normalizedScore: number;
+  failurePercentage: number;
+  successPercentage: number;
+}
+
+// Basic type guard for array of FailedTransactionsCorrelation
+export const isFailedTransactionsCorrelations = (
+  arg: unknown
+): arg is FailedTransactionsCorrelation[] => {
+  return Array.isArray(arg) && arg.length > 0;
```

Comment:
I think this should also check if arg[0] matches the signature of `FailedTransactionsCorrelation`

Reply:
FYI Decided the type guard was overkill and changed it back to inline checks in c52158b89a0.